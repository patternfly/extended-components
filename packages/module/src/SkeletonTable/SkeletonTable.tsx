import React, { ReactNode } from 'react';
import {
  Caption,
  Table,
  TableProps,
  TableVariant,
  RowSelectVariant,
  ThProps
} from '@patternfly/react-table';
import SkeletonTableBody, { SkeletonTableBodyProps } from '../SkeletonTableBody';
import SkeletonTableHead, { SkeletonTableHeadProps } from '../SkeletonTableHead';

export interface SkeletonTableProps extends Omit<TableProps, 'ref'>, Omit<SkeletonTableBodyProps, 'columnsCount'>, SkeletonTableHeadProps {
  /** Indicates the table variant */
  variant?: TableVariant;
  /** Flag indicating if the table should have borders */
  borders?: boolean;
  /** The number of rows the skeleton table should contain */
  rowsCount?: number;
  /** Any captions that should be added to the table */
  caption?: React.ReactNode;
  /** Custom OUIA ID */
  ouiaId?: string | number;
  /** Flag indicating if the table is selectable */
  isSelectable?: boolean;
  /** Flag indicating if the table is expandable */
  isExpandable?: boolean;
  /** Determines if the row selection variant (radio/checkbox) */
  selectVariant?: RowSelectVariant;
  /** Custom columns for the table */
  columns?: (ReactNode | { cell: ReactNode; props?: ThProps })[];
  /** Number of columns in the table */
  columnsCount?: number;
}

const SkeletonTable: React.FunctionComponent<SkeletonTableProps> = ({
  variant,
  borders = true,
  rowsCount = 5,
  caption,
  ouiaId = 'SkeletonTable',
  isSelectable,
  isExpandable,
  selectVariant = RowSelectVariant.checkbox,
  columns,
  columnsCount,
  isTreeTable,
  ...rest
}: SkeletonTableProps) => {
  const rowCellsCount = Array.isArray(columns) ? columns.length : columnsCount;

  return (
    <Table aria-label="Loading" variant={variant} borders={borders} ouiaId={ouiaId} {...rest}>
      {caption && <Caption>{caption}</Caption>}
      <SkeletonTableHead 
        ouiaId={ouiaId}
        isSelectable={isSelectable}
        isExpandable={isExpandable}
        columnsCount={columnsCount}
        columns={columns}
        isTreeTable={isTreeTable}
      />
      <SkeletonTableBody 
        columnsCount={rowCellsCount ?? 0}
        rowsCount={rowsCount}
        isSelectable={isSelectable}
        isExpandable={isExpandable}
        selectVariant={selectVariant}
        ouiaId={ouiaId}
      />
    </Table>
  );
};

export default SkeletonTable;
