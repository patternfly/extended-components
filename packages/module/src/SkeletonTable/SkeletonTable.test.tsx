import React from 'react';
import { render } from '@testing-library/react';
import SkeletonTable from './SkeletonTable';

describe('SkeletonTable component', () => {
  it('should render correctly', () => {
    expect(render(<SkeletonTable columns={[ 'First', 'Second' ]}/>)).toMatchSnapshot();
  });

  it('should render correctly with rows', () => {
    expect(render(<SkeletonTable columns={[ 'First', 'Second' ]} rows={10} />)).toMatchSnapshot();
  });
});
