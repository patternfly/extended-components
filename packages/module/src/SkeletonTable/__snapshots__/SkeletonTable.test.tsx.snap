// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonTable component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        aria-label="Loading"
        class="pf-v6-c-table pf-m-grid-md"
        data-ouia-component-id="SkeletonTable"
        data-ouia-component-type="PF6/Table"
        data-ouia-safe="true"
        role="grid"
      >
        <thead
          class="pf-v6-c-table__thead"
          data-ouia-component-id="SkeletonTable-thead"
        >
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-head"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <th
              class="pf-v6-c-table__th"
              data-ouia-component-id="SkeletonTable-th-0"
              scope="col"
              tabindex="-1"
            >
              first
            </th>
            <th
              class="pf-v6-c-table__th"
              data-ouia-component-id="SkeletonTable-th-1"
              scope="col"
              tabindex="-1"
            >
              second
            </th>
          </tr>
        </thead>
        <tbody
          class="pf-v6-c-table__tbody"
          data-ouia-component-id="SkeletonTable-tbody"
          role="rowgroup"
        >
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-0"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-0-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-0-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-1"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-1-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-1-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-2"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-2-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-2-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-3"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-3-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-3-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-4"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-4-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-4-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      aria-label="Loading"
      class="pf-v6-c-table pf-m-grid-md"
      data-ouia-component-id="SkeletonTable"
      data-ouia-component-type="PF6/Table"
      data-ouia-safe="true"
      role="grid"
    >
      <thead
        class="pf-v6-c-table__thead"
        data-ouia-component-id="SkeletonTable-thead"
      >
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-head"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <th
            class="pf-v6-c-table__th"
            data-ouia-component-id="SkeletonTable-th-0"
            scope="col"
            tabindex="-1"
          >
            first
          </th>
          <th
            class="pf-v6-c-table__th"
            data-ouia-component-id="SkeletonTable-th-1"
            scope="col"
            tabindex="-1"
          >
            second
          </th>
        </tr>
      </thead>
      <tbody
        class="pf-v6-c-table__tbody"
        data-ouia-component-id="SkeletonTable-tbody"
        role="rowgroup"
      >
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-0"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-0-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-0-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-1"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-1-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-1-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-2"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-2-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-2-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-3"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-3-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-3-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-4"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-4-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-4-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SkeletonTable component should render correctly with rows 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        aria-label="Loading"
        class="pf-v6-c-table pf-m-grid-md"
        data-ouia-component-id="SkeletonTable"
        data-ouia-component-type="PF6/Table"
        data-ouia-safe="true"
        role="grid"
      >
        <thead
          class="pf-v6-c-table__thead"
          data-ouia-component-id="SkeletonTable-thead"
        >
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-head"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <th
              class="pf-v6-c-table__th"
              data-ouia-component-id="SkeletonTable-th-0"
              scope="col"
              tabindex="-1"
            >
              first
            </th>
            <th
              class="pf-v6-c-table__th"
              data-ouia-component-id="SkeletonTable-th-1"
              scope="col"
              tabindex="-1"
            >
              second
            </th>
          </tr>
        </thead>
        <tbody
          class="pf-v6-c-table__tbody"
          data-ouia-component-id="SkeletonTable-tbody"
          role="rowgroup"
        >
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-0"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-0-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-0-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-1"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-1-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-1-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-2"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-2-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-2-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-3"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-3-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-3-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v6-c-table__tr"
            data-ouia-component-id="SkeletonTable-tr-4"
            data-ouia-component-type="PF6/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-4-0"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v6-c-table__td"
              data-ouia-component-id="SkeletonTable-td-4-1"
              tabindex="-1"
            >
              <div
                class="pf-v6-c-skeleton"
              >
                <span
                  class="pf-v6-screen-reader"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      aria-label="Loading"
      class="pf-v6-c-table pf-m-grid-md"
      data-ouia-component-id="SkeletonTable"
      data-ouia-component-type="PF6/Table"
      data-ouia-safe="true"
      role="grid"
    >
      <thead
        class="pf-v6-c-table__thead"
        data-ouia-component-id="SkeletonTable-thead"
      >
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-head"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <th
            class="pf-v6-c-table__th"
            data-ouia-component-id="SkeletonTable-th-0"
            scope="col"
            tabindex="-1"
          >
            first
          </th>
          <th
            class="pf-v6-c-table__th"
            data-ouia-component-id="SkeletonTable-th-1"
            scope="col"
            tabindex="-1"
          >
            second
          </th>
        </tr>
      </thead>
      <tbody
        class="pf-v6-c-table__tbody"
        data-ouia-component-id="SkeletonTable-tbody"
        role="rowgroup"
      >
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-0"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-0-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-0-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-1"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-1-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-1-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-2"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-2-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-2-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-3"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-3-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-3-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTable-tr-4"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-4-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTable-td-4-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
